From 0eca520e564b111cad081f5018bcdfb1e8b61a73 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 2 Sep 2024 15:17:28 +0800
Subject: [PATCH 21/21] hack(buildtools): revert upstream check for FFmpeg
 versions

Reverts 855806fefdd52b29e8b15b6a02e263afc21028c8:

  Add check for system ffmpeg compatibility

  Chromium uses the first_dts field from AVStream that has
  been moved out from public api in ffmpeg 5.0.
  Although some	packagers patch their ffmpeg to be compatible
  with chromium.
  Add compile time check to test compatibility.

  Task-number: QTBUG-116553
  Change-Id: I7658b9b12cb5122b6485f063edc3280f31fe9273
  Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
  (cherry picked from commit 7e46ff40)

  Reviewed-by: Michal Klocek <michal.klocek@qt.io>
---
 qtwebengine/src/buildtools/configure.json | 26 -----------------------
 1 file changed, 26 deletions(-)

diff --git a/qtwebengine/src/buildtools/configure.json b/qtwebengine/src/buildtools/configure.json
index 9e7a0c5704..88d1790c17 100644
--- a/qtwebengine/src/buildtools/configure.json
+++ b/qtwebengine/src/buildtools/configure.json
@@ -212,27 +212,6 @@
                 { "type": "pkgConfig", "args": "libavcodec libavformat libavutil" }
             ]
         },
-        "webengine-ffmpeg-support": {
-            "label": "compatible ffmpeg",
-            "type": "compile",
-            "test": {
-                "head": [
-                    "#include <libavformat/version.h>",
-                    "extern \"C\" {",
-                    "#include <libavformat/avformat.h>",
-                    "}"
-                ],
-                "main": [
-                    "#if LIBAVFORMAT_VERSION_MAJOR >= 59",
-                        "AVStream stream;",
-                        "auto first_dts = av_stream_get_first_dts(&stream);",
-                    "#endif"
-                ]
-            },
-            "sources": [
-                { "type": "pkgConfig", "args": "libavformat" }
-            ]
-        },
         "webengine-opus": {
             "label": "opus",
             "sources": [
@@ -779,11 +758,6 @@
             "type": "warning",
             "condition": "config.ios && config.simulator && config.device && features.build-qtpdf",
             "message": "Building fat libray with device and simulator architectures will disable NEON."
-        },
-        {
-            "type": "fatal",
-            "condition": "features.webengine-system-ffmpeg && !libs.webengine-ffmpeg-support",
-            "message": "Unmodified ffmpeg >= 5.0 is not supported. Please configure with -qt-webengine-ffmpeg."
         }
     ],
     "summary": [
-- 
2.46.0

